docs/build/build-options.md

```diff
@@ -421,6 +422,55 @@ to use either of those (`ENABLE_STDCXX_SYNC` excludes `ENABLE_MONOTONIC_CLOCK`).
 When ON, this option enables the standard C++ `thread` and `chrono` libraries 
 (available since C++11) to be used by SRT instead of the `pthreads` libraries.
 
+#### ENABLE_PKTINFO
+**`--enable-pktinfo`** (default: OFF)
+
+This allows the use of `IP_PKTINFO` control message which can be used to
+extract the true target IP address from the incoming UDP packets to a listener
+bound to a so-called "any" address (`0.0.0.0` for IPv4, AKA `INADDR_ANY`, or
+`::` for IPv6, AKA `in6addr_any`, or simply when the application clears the
+`sockaddr*` structure and only sets the port number), and then set this address
+as source IP address when sending packets to the peer. This solves the problem,
+when the routing rules in the agent's host make that sending a packet to a
+given IP address would use a different source IP address than the one that was
+set as a target IP address in the UDP packet from the peer. The peer will
+reject such a packet as a suspected MITM attempt, and this will lead to a
+connection failure.
```

Comment:
```suggestion
This allows the use of the `IP_PKTINFO` control message to extract the true target IP address from the incoming 
UDP packets to a listener bound to "any" address. For IPv4, this corresponds to 0.0.0.0 (or `INADDR_ANY`), and 
for IPv6 it corresponds to `::` (or `in6addr_any`). This also happens when the application clears the `sockaddr*` 
structure and only sets the port number. This true address can then be specified as the source IP address when 
sending packets to the peer. This solves the problem where routing rules in an agent's host send a packet using a 
different source IP address than the  target IP address in the UDP packet from the peer. The peer will reject such a 
packet as a suspected man-in-the-middle (MITM) attempt, and this will lead to a connection failure.
```

Reply:
@ethouris Just to be clear, on line 429 "...packets to a listener bound to "any" address...", the word "bound" refers to the packets themselves, not the listener, right?

Reply:
Well, I don't exactly like this. The intention of the beginning of this description was to define what this "any" address means, as the problem concerns only listener sockets that are bound to "any" address. Maybe something like this:

... incoming UDP packets to a listener socket bound to "any" address. The "any" address is defined in IPv4 as `0.0.0.0` (`INADDR_ANY`) and in IPv6 as `::` (`in6addr_any`), and the applications usually make it by clearing the `sockaddr*` structure and only setting the port number.

Also "specified" - I don't think it is precise enough. The UDP packets always have source and destination address. in the headers, and these are set in this packet by the system. The CMSG message allows to override this setting with a desired value. This mechanism relies on that first the target address is read from the incoming packet, and then this same address is set as source address to the outgoing packets.

Reply:
```suggestion
This allows the use of the `IP_PKTINFO` control message to extract the true target IP 
address from the incoming UDP packets to a listener bound to "any" address. The "any" 
address is defined in IPv4 as 0.0.0.0 (`INADDR_ANY`) and in IPv6 as :: (`in6addr_any`). 
Applications usually implement it by clearing the `sockaddr*` structure and only setting 
the port number. This true address can then be used to override the source IP address 
when sending packets to the peer. This solves the problem where routing rules 
in an agent's host send a packet using a different source IP address than the  target 
IP address in the UDP packet from the peer. The peer will reject such a packet as a 
suspected man-in-the-middle (MITM) attempt, leading to a connection failure.
```

Reply:
Perfect.